lib/LanguagePicker.js

```diff
@@ -19,50 +42,88 @@ function LanguagePicker(lang = 'en', config = {}) {
 
   // Add prefix to all languages if exists
   // eslint-disable-next-line arrow-body-style
   fallbacks = fallbacks.map((code) => {
     return code === this.nameTag ? code : this.prefix + code;
   });
 
+  // Store initial fallbacks
   this.fallbacks = fallbacks;
 }
 
 LanguagePicker.prototype.newProcessor = function newProcessor() {
-  const { fallbacks, nameTag, prefix } = this;
-  let bestChoiceLang = nameTag || (prefix ? `${prefix}en` : 'en');
-  let bestChoiceValue;
   let firstChoiceValue;
 
   return {
-    addValue(lang, value) {
-      const langIndex = fallbacks.indexOf(lang);
-      if (
-        langIndex > -1 &&
-        langIndex <= fallbacks.indexOf(bestChoiceLang)
-      ) {
-        // Given language is a better fallback
-        bestChoiceLang = lang;
-        bestChoiceValue = value;
+    addValue: (lang, value) => {
+      this.values[lang] = value;
+
+      if (!firstChoiceValue) {
+        firstChoiceValue = value;
+      }
+    },
+    getResult: () => {
+      let val;
+
+      // Get the best value from the best language fallback:
+      // 1. Requested language
+      if (this.values[this.userLang]) {
+        return this.values[this.userLang];
+      }
+
+      // 2. Fallback language from fallbacks.json
+      for (const fallback of this.fallbacks) {
+        val = this.values[fallback];
+        if (val) {
+          return val;
+        }
+      }
+
+      // 3. Any language with suffix of same script as requested language
+      const valueLangCodes = Object.keys(this.values);
+      for (const langCode of valueLangCodes) {
+        val = this.values[langCode];
+        if (val && langCode.endsWith(this.langScript)) {
+          return val;
+        }
+      }
+
+      // 4. Another language with the same script
+      for (const langCode of this.languagesInScript) {
+        val = this.values[langCode];
+        if (val) {
+          return val;
+        }
+      }
+
+      // 5. English
+      val = this.values[this.prefix ? `${this.prefix}en` : 'en'];
+      if (val) {
+        return val;
+      }
+
+      // 6. Any language with suffix 'Latn'
+      // Only do this if the script isn't already latin;
+      // otherwise, we've done it in step 3
+      for (const value of valueLangCodes) {
+        if (value && value.endsWith('Latn')) {
+          return this.values[value];
+        }
       }
 
-      // Store the value of the first value given
-      firstChoiceValue = firstChoiceValue !== undefined ?
-        firstChoiceValue : value;
+      // If nothing is found:
+      // - If there's a name tag, return it
+      // - If there's no name tag, return first choice
+      return this.nameTag && this.values[this.nameTag] ?
+        this.values[this.nameTag] :
+        firstChoiceValue;
     },
-    getResult: () => bestChoiceValue ||
-        (bestChoiceLang && fallbacks[bestChoiceLang]) ||
-        firstChoiceValue,
+    // This is for testing purposes
+    getFallbacks: () => this.fallbacks,
```

Comment:
I'm looking at the tests but I'm not sure where this is being used...

Reply:
It was on a previous iteration, I forgot to remove this